corehq/apps/fixtures/upload/workbook.py

```diff
@@ -70,20 +72,27 @@ def iter_tables(self, domain):
                 item_attributes=sheet.item_attributes,
             )
 
-    def iter_rows(self, data_type):
+    def iter_rows(self, data_type, sort_keys):
         data_items = list(self.get_data_sheet(data_type.tag))
         type_fields = data_type.fields
-        for sort_key, di in enumerate(data_items):
-            yield FixtureDataItem(
+        sort_key = -1
+        for i, di in enumerate(data_items):
+            sort_key = max(sort_keys.get(di['UID'], i), sort_key + 1)
```

Comment:
`sort_keys` is a dict of `{<row UUID>: <old sort key (int)>}`

> Is it correct to say it was previously sorted by the order of the rows in the sheet and now is sorted by the unique id of the row, which means it was user controlled and now is not?

No. It is still user-controlled, and now it is more correct than it was before because the order of the rows in the spreadsheet always determine the sort order where before that was only true when using `replace=True` (in which case all rows were replaced with new rows).

Previously, uploading with `replace=False` would cause new rows to be assigned sort keys that matched existing rows, which would cause old and new rows to be intermixed.